### PR TITLE
Read many fix

### PR DESCRIPTION
**Pull request by jeet1995**
# Description

This PR fixes the issue when a point-read based `404 / 0` response drops existing point-read based documents from the resultant `FeedResponse` instance in spite of the ID of these documents being specified in the `readMany` request. The expected behavior is to ignore non-existent documents but fetch other documents.

# Testing

Test: `CosmosItemTest#readManyWithMultiplePartitionsAndSome404s`

- Create a multi-partition container - preferably with more than 2 physical partitions.
- Ensure documents are created on each physical partition.
- Query 1 document per physical partition using the `FeedRange` instance for the physical partition. This is done to ensure when `readMany` is used to query the documents back, it only uses point-reads to do so.
- Choose _n_ documents in which a non-existent ID is injected.
- Read these documents back using the `readMany` API.
- Verify if the count of fetched documents through `readMany` equals (total no. of physical partitions) - _n_  
- Repeat the above steps by varying _n_ from 0 to the total no. of physical partitions.

# Benchmarking results

- The fix shows a marginal tail latency increase (in the order of 1 ms to 5 ms)
image

### CODE REVIEW

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java [hunk 1]
```diff
@@ -354,84 +361,107 @@ public void readManyWithManyNonExistentItemIds() throws Exception {
...
-            String documentId = UUID.randomUUID().toString();
-            ObjectNode document = getDocumentDefinition(documentId, partitionKeyValue);
-            container.createItem(document);
-        }
+        try {
 
-        // query 1 item
-        SqlQuerySpec sqlQuerySpec = new SqlQuerySpec();
-        StringBuilder stringBuilder = new StringBuilder();
+            readManyClient = new CosmosClientBuilder()
```
[the remark this answers is not in the document]
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:375]: This is a miss - fixed in next iteration.

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java [hunk 2]
```diff
@@ -55,6 +61,7 @@ public class CosmosItemTest extends TestSuiteBase {
         ImplementationBridgeHelpers.CosmosDiagnosticsHelper.getCosmosDiagnosticsAccessor();
 
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Random random = new Random();
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:64]: Random is not thread-safe (neither in Java nor .Net) - please use ThreadLocalRandom.current instead

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java:64]: Not really critical in test code - but there is always the risk that code is copied and pasted - so, better to fix this. Non-blocking comment thought.

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM - Thanks

File: sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java (hunk 2 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosItemTest.java]: Fixed in next iteration.

**Comment by jeet1995** [comment]
There are some metric-related tests which don't pass - would require some looking at. The CosmosConflictTests on the other passes locally.

**Pull request merged**